src/goto-programs/string_refine_preprocess.cpp

```diff
@@ -0,0 +1,1271 @@
+  side_effect_functions
+    ["java::java.lang.StringBuilder.delete:(II)Ljava/lang/StringBuilder;"]=
+    ID_cprover_string_delete_func;
+  side_effect_functions
+    ["java::java.lang.StringBuilder.deleteCharAt:(I)Ljava/lang/StringBuilder;"]=
+    ID_cprover_string_delete_char_at_func;
+  side_effect_functions
+    ["java::java.lang.StringBuilder.insert:(ILjava/lang/String;)"
+     "Ljava/lang/StringBuilder;"]=
+    ID_cprover_string_insert_func;
+  side_effect_functions
+    ["java::java.lang.StringBuilder.insert:(II)Ljava/lang/StringBuilder;"]=
+    ID_cprover_string_insert_int_func;
+  side_effect_functions
+    ["java::java.lang.StringBuilder.insert:(IJ)Ljava/lang/StringBuilder;"]=
+    ID_cprover_string_insert_long_func;
+  side_effect_functions
+    ["java::java.lang.StringBuilder.insert:(IC)Ljava/lang/StringBuilder;"]=
+    ID_cprover_string_insert_char_func;
+  side_effect_functions
+    ["java::java.lang.StringBuilder.insert:(IZ)Ljava/lang/StringBuilder;"]=
+    ID_cprover_string_insert_bool_func;
+  side_effect_functions
+    ["java::java.lang.StringBuilder.setLength:(I)V"]=
+    ID_cprover_string_set_length_func;
+
+
+  side_effect_char_array_functions
+    ["java::java.lang.StringBuilder.insert:(I[CII)Ljava/lang/StringBuilder;"]=
+    ID_cprover_string_insert_char_array_func;
+  side_effect_char_array_functions
+    ["java::java.lang.StringBuilder.insert:(I[C)Ljava/lang/StringBuilder;"]=
+    ID_cprover_string_insert_char_array_func;
+
+  string_function_calls
+    ["java::java.lang.String.<init>:(Ljava/lang/String;)V"]=
+    ID_cprover_string_copy_func;
+  string_function_calls
+    ["java::java.lang.String.<init>:(Ljava/lang/StringBuilder;)V"]=
+    ID_cprover_string_copy_func;
+  string_function_calls
+    ["java::java.lang.StringBuilder.<init>:(Ljava/lang/String;)V"]=
+    ID_cprover_string_copy_func;
+  string_function_calls["java::java.lang.String.<init>:()V"]=
+    ID_cprover_string_empty_string_func;
+  string_function_calls["java::java.lang.StringBuilder.<init>:()V"]=
+    ID_cprover_string_empty_string_func;
+
+  string_of_char_array_function_calls["java::java.lang.String.<init>:([C)V"]=
+    ID_cprover_string_of_char_array_func;
+  string_of_char_array_function_calls["java::java.lang.String.<init>:([CII)V"]=
+    ID_cprover_string_of_char_array_func;
+
+  string_of_char_array_functions
+    ["java::java.lang.String.valueOf:([CII)Ljava/lang/String;"]=
+    ID_cprover_string_of_char_array_func;
+  string_of_char_array_functions
+    ["java::java.lang.String.valueOf:([C)Ljava/lang/String;"]=
+    ID_cprover_string_of_char_array_func;
+  string_of_char_array_functions
+    ["java::java.lang.String.copyValueOf:([CII)Ljava/lang/String;"]=
+    ID_cprover_string_of_char_array_func;
+  string_of_char_array_functions
+    ["java::java.lang.String.copyValueOf:([C)Ljava/lang/String;"]=
+    ID_cprover_string_of_char_array_func;
+
+  c_string_functions["__CPROVER_uninterpreted_string_literal_func"]=
```

Comment:
It only differs in the `uninterpreted` part, but these names are quite arbitrary, I kept this map to be compatible with the tests written by Alberto but ultimately this should be replaced by the actual name of C string functions (strcpy, strlen, etc...)